pkg/kvstore/etcd.go

```diff
@@ -432,9 +433,11 @@ func (e *etcdClient) isConnectedAndHasQuorum(ctx context.Context) error {
 
 	select {
 	// Wait for the the initial connection to be established
-	case err := <-e.firstSession:
-		if err != nil {
-			return err
+	case <-e.firstSession:
+		if err := e.sessionError(); err != nil {
+			if err != nil {
+				return err
+			}
```

Comment:
Is the nested `nil` check still required?

Reply:
Good spotting.